alchemist/alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/ScafiIncarnation.scala

```diff
@@ -45,12 +44,36 @@ sealed class ScafiIncarnation[P <: Position[P]] extends Incarnation[Any, P]{
 
   override def createAction(
       rand: RandomGenerator,
-      env: Environment[Any, P],
-      node: Node[Any],
-      time: TimeDistribution[Any],
-      reaction: Reaction[Any],
-      param: String) = {
-    new RunScafiProgram(notNull(env), notNull(node), notNull(reaction), notNull(rand), notNull(param))
+      env: Environment[T, P],
+      node: Node[T],
+      time: TimeDistribution[T],
+      reaction: Reaction[T],
+      param: String
+  ) = {
+    if (!node.isInstanceOf[ScafiNode[T,P]]) {
+      throw new IllegalStateException(getClass.getSimpleName + " cannot get cloned on a node of type " + node.getClass.getSimpleName)
+    }
+    val scafiNode = node.asInstanceOf[ScafiNode[T,P]]
+    if(param=="send") {
+      val alreadyDone = ScafiIncarnationUtils.allActions[T,P,SendScafiMessage[T,P]](node, classOf[SendScafiMessage[T,P]]).map(_.program)
+        // ScafiIncarnationUtils.allScafiProgramsFor[T,P](node).filter(_.isComputationalCycleComplete)
+      val spList = ScafiIncarnationUtils.allScafiProgramsFor[T,P](node) -- alreadyDone
+
```

Comment:
Remove empty lines within methods

Reply:
Or, if you think that some spacing is useful, write a comment instead explaining why :)